packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,341 @@
+   * In that case, each index child of this job must wait for the corresponding
+   * index child of each dependency to complete before it can begin.
+   *
+   * @default - No dependencies
+   */
+  readonly dependsOn?: JobDependency[];
+
+  /**
+   * The payload to be passed as parametrs to the batch job
+   *
+   * @default - No parameters are passed
+   */
+  readonly payload?: { [key: string]: any };
+
+  /**
+   * The retry strategy to use for failed jobs from this SubmitJob operation.
+   * When a retry strategy is specified here,
+   * it overrides the retry strategy defined in the job definition.
+   *
+   * @default - No retry strategy
+   */
+  readonly retryStrategy?: RetryStrategy;
+
+  /**
+   * The timeout configuration for this SubmitJob operation.
+   *
+   * @default - No timeout
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The service integration pattern indicates different ways to call TerminateCluster.
+   *
+   * The valid value is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default SYNC
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+}
+
+/**
+ * A Step Functions Task to run AWS Batch
+ *
+ * @experimental
+ */
+export class RunBatchJob implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly props: RunBatchJobProps) {
+    this.integrationPattern =
+      props.integrationPattern || sfn.ServiceIntegrationPattern.SYNC;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    return {
+      resourceArn: getResourceArn(
+        'batch',
+        'submitJob',
+        this.integrationPattern
+      ),
+      policyStatements: [
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: [
+            'batch:SubmitJob',
+            'batch:DescribeJobs',
+            'batch:TerminateJob'
+          ]
+        }),
+        new iam.PolicyStatement({
+          resources: [
+            `arn:aws:events:${Aws.REGION}:${Aws.ACCOUNT_ID}:rule/StepFunctionsGetEventsForBatchJobsRule`
+          ],
+          actions: [
+            'events:PutTargets',
+            'events:PutRule',
+            'events:DescribeRule'
+          ]
+        })
+      ],
+      parameters: {
+        JobDefinition: this.props.jobDefinition.jobDefinitionArn,
+        JobName: this.props.jobName,
+        JobQueue: this.props.jobQueue.jobQueueArn,
+        Parameters: this.props.payload,
+
+        ...(this.props.arrayProperties && {
+          ArrayProperties: {
+            Size: this.props.arrayProperties.size
+          }
+        }),
```

Comment:
Can you explain what you're doing here?

Wouldn't `arrayProperties: this.props.arrayProperties` achieve the same?

Reply:
I am trying to parse the props to the appropriate syntax since the [docs](https://docs.aws.amazon.com/step-functions/latest/dg/connect-batch.html) mentions:
> Parameters in Step Functions are expressed in PascalCase, even when the native service API is camelCase.

Reply:
This should be automatically taken care of. Everything from the CDK should get transformed to `PascalCase` when it's converted to CloudFormation. 
Can you test this out and report back?

Reply:
I think I started off that way but since these are nested Parameter objects, I think CFN ignore them. But I will give it a go again.

Reply:
Just verified by unit-tests that these nested parameters **does not** get converted to `PascalCase`